core/test/mpi/distributed/dense_communicator.cpp

```diff
@@ -0,0 +1,243 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/distributed/dense_communicator.hpp>
+
+#include "core/test/utils/assertions.hpp"
+
+using gko::experimental::mpi::comm_index_type;
+
+class DenseCommunicator : public ::testing::Test {
```

Comment:
the tests between DenseCommunicator and NeighborCommunicator seem to be the same.
maybe handle it via TypeTest?